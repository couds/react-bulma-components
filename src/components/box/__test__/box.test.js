import React from 'react';
import renderer from 'react-test-renderer';
import Box from '..';

describe('Box component', () => {
  it('Should Exist', () => {
    expect(Box).toMatchSnapshot();
  });
  it('Should have box classname', () => {
    const component = renderer.create(
      <Box>Facebook</Box>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat Bulma class with classes in props', () => {
    const component = renderer.create(
      <Box className="other-class test">Facebook</Box>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render as an html section', () => {
    const component = renderer.create(
      <Box renderAs="section">This should be a section</Box>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have custom inline styles', () => {
    const component = renderer.create(
      <Box renderAs="section" style={{ width: 200, zIndex: 1 }}>This should be a section with custom styles</Box>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
