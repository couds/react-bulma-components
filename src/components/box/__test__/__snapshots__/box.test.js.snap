// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component Should Exist 1`] = `[Function]`;

exports[`Box component Should concat Bulma class with classes in props 1`] = `
<div
  className="box other-class test"
  style={Object {}}
>
  Facebook
</div>
`;

exports[`Box component Should have box classname 1`] = `
<div
  className="box"
  style={Object {}}
>
  Facebook
</div>
`;

exports[`Box component Should have custom inline styles 1`] = `
<section
  className="box"
  style={
    Object {
      "width": 200,
      "zIndex": 1,
    }
  }
>
  This should be a section with custom styles
</section>
`;

exports[`Box component Should render as an html section 1`] = `
<section
  className="box"
  style={Object {}}
>
  This should be a section
</section>
`;
